libavcodec/vaapi_decode.c

```diff
@@ -569,6 +587,22 @@ static int vaapi_decode_make_config(AVCodecContext *avctx,
         frames->width = avctx->coded_width;
         frames->height = avctx->coded_height;
 
+        if (avctx->sub_frame_opts) {
+            AVDictionaryEntry *e = NULL;
+            while ((e = av_dict_get(avctx->sub_frame_opts, "", e, AV_DICT_IGNORE_SUFFIX))) {
+                if (!strcmp(e->key, "width"))
+                    frames->sfc_w = atoi(e->value);
+                else if (!strcmp(e->key, "height"))
+                    frames->sfc_h = atoi(e->value);
+                else if (!strcmp(e->key, "format"))
+                    frames->sfc_format = av_get_pix_fmt(e->value);
+            }
+        }
```

Comment:
If the user only set height and width will be 0, Is this configure still acceptable by libva? Or we can set default sfc_w/h equals to original w/h?